services/nexus/nexus.service.js

```diff
@@ -16,7 +16,10 @@ const searchApiSchema = Joi.object({
       Joi.object({
         latestRelease: optionalDottedVersionNClausesWithOptionalSuffix,
         latestSnapshot: optionalDottedVersionNClausesWithOptionalSuffix,
-        version: optionalDottedVersionNClausesWithOptionalSuffix,
+        version: Joi.alternatives().try(
+          optionalDottedVersionNClausesWithOptionalSuffix,
+          Joi.string()
```

Comment:
Given `optionalDottedVersionNClausesWithOptionalSuffix` is a special case of `string()`, I don't think we're actually gaining anything here by testing against `optionalDottedVersionNClausesWithOptionalSuffix` first if we'll fall back to accepting any string anyway.

Also should it be `Joi.string().required()`?

Reply:
> Also should it be Joi.string().required()?

No I don't think it can be required. I will review some previous issues we had with this schema to find out defnitively, but I believe these fields were set to optional intentionally because they are not always present on each object in the response array. 

> Given optionalDottedVersionNClausesWithOptionalSuffix is a special case of string(), I don't think we're actually gaining anything here by testing against optionalDottedVersionNClausesWithOptionalSuffix first if we'll fall back to accepting any string anyway.

That's a logical point. I do think the scenario of one of the previously published versions for an artifact having a version that doesn't include a digit is much more the exception than the norm (to my knowledge no one has reported the issue I saw while working on this).

I do think there's value in a future reader looking at this code and understanding that this field is generally expected to follow the same version pattern, but that there are a couple exceptional circumstances that necessitate a less-strict option. 

Perhaps it would be better change the schema to `Joi.string()` as you suggest, and include an inline comment to convey the above rationale?

Reply:
Right yeah clue's in the name **optional**DottedVersion.. :)

I do see your point on making it clear that `version` will usually follow the same pattern as `latestRelease`/`latestSnapshot` (except when it doesn't). I think I have a slight preference for just explaining it in a comment, but its fine to present that in either of the ways you've suggested really.